### PR TITLE
Bump Ruby from 3.0.3 to 3.1.4

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

I forgot that we were still using Ruby 3.0 and encountered a syntax error while writing valid Ruby 3.1 code. I would like to avoid that happening again. :wink: 

Major 3.1 highlights:

* Performance improvements.
* Better autocompletion in IRB.
* Keyword arguments of the same name can be omitted.
* https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/

#### What should we test?

This has the potential to break anything but we can probably just do a sanity check:

- Go through the checkout.
- Any parts not covered by automated tests?

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by jibees — APPROVED** [review]
👌 

I'm wondering if we can spot any new deprecation warnings that could occurs through all the specs?

File: Gemfile [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
 
 source 'https://rubygems.org'
-ruby "3.0.3"
 git_source(:github) { |repo_name| "https://github.com/#{repo_name}.git" }
 
+ruby File.read('.ruby-version').chomp
```
Comment (dacook) [inline Gemfile:6]: 🏅

**Review by dacook — APPROVED** [review]
Yay, seems fine!

> I'm wondering if we can spot any new deprecation warnings that could occurs through all the specs?

Good question. I did a quick spot-check of about 6 different types of tests and couldn't see any warnings in the logs.

**Comment by dacook** [comment]
Deploy failed: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/au-staging/deploys/668

I thought that rbenv would automatically download the necessary version. Do we need to trigger that separately?

**Comment by mkllnk** [comment]
It's currently managed in ofn-install:
https://github.com/openfoodfoundation/ofn-install/blob/51fc06623d824e2e0443938ffedb5d4cc8b1d356/inventory/group_vars/all.yml#L60-L66

Maybe we should move that role to the deployment instead of provisioning. Not sure how long it takes. Otherwise using the rbenv-install script instead and trigger it during deployment could be a good option. I don't really want to get distracted with that right now...

**Comment by drummer83** [comment]
Hi @mkllnk,
Deployment worked well (still using semaphore).
I also did a quick test covering most parts of our release test procedure (except for the race conditions) because those are the things which are not covered by automated tests.
It's all looking good.

I'll move this to Ready To Go but won't merge myself because we have failures regarding the Github deployment. I guess they are not relevant, but not 100 % sure.

Thanks!

**Pull request merged**